### ISSUE TITLE
 Feature Request - Replace External URL with Local File for XML References

### ISSUE THREAD

**Issue by jose-mc-b** [microsoft/XmlNotepad#412]
I would like to request a new feature that allows users to replace external URLs (e.g., http/https) with local files when opening XML documents. This would be particularly useful for XML files containing external references, as it would prevent the need to fetch these resources from the web each time the file is opened. Instead, users would be able to specify a local file path to be used in place of the external URL. 
![xmlnotepad](https://github.com/user-attachments/assets/6a92ae1e-7f2a-48ce-82ef-ded74ab87d7d)

**Comment by lovettchris** [comment]
Yes, I would call this a "schema caching" feature. How about we let you specify a folder (defaults to someplace in your TEMP directory) and a "check for updates" frequency that allows newer versions of the schema to be downloaded at some point, where that update frequency could be set to something very large if you never want to check for updates.  The code already contains an in-memory "SchemaCache" to avoid multiple downloads while you are editing, so all we need to do is add a backing storage folder to this "cache" and we'd have what you want I think.  Might also need some sort of cleanup feature that deletes old stale schemas you haven't used in a long time so the folder doesn't grow in unbounded size...
